src/run/run.c

```diff
@@ -1496,11 +1496,13 @@ static int on_properties_changed(sd_bus_message *m, void *userdata, sd_bus_error
 }
 
 static int pty_forward_handler(PTYForward *f, int rcode, void *userdata) {
-        RunContext *c = userdata;
+        RunContext *c = ASSERT_PTR(userdata);
 
         assert(f);
 
-        if (rcode < 0) {
+        if (rcode == -ECANCELED)
+                log_debug_errno(rcode, "PTY forwarder disconnected.");
```

Comment:
hmm, so you dropped the exiting of the event loop here?

is that intentional? commit msg doesn't mention this?